### ISSUE TITLE
Chronos' build process differs from other components

### ISSUE THREAD

**Issue by dtheobald** [Metaswitch/chronos#215]
After a clean clone and checking out release-88, I get the following when trying to build chronos:
```
$ make deb
g++   -MMD -MP -O2 -ggdb3 -std=c++11 -Wall -Werror -Isrc/include -Imodules/cpp-common/include -Imodules/rapidjson/include -Imodules/sas-client/include -Ibuild/usr/include -c src/main/chronos_internal_connection.cpp -o build/chronos/chronos_internal_connection.o
In file included from src/main/chronos_internal_connection.cpp:45:0:
modules/sas-client/include/sas.h:42:20: fatal error: config.h: No such file or directory
#include <config.h>
                    ^
compilation terminated.
make: *** [build/chronos/chronos_internal_connection.o] Error 1
```
To fix this I had to make this change:
```
diff --git a/Makefile b/Makefile
index f4e5491..03d4699 100644
--- a/Makefile
+++ b/Makefile
@@ -49,8 +49,11 @@ include $(patsubst %, mk/%.mk, ${SUBMODULES})
 submodules :
 	${MAKE} ${SUBMODULES}
 
+.PHONY: build
+build: ${SUBMODULES} chronos
+
 .PHONY: deb
-deb: chronos
+deb: build
 	${MAKE} deb-only
 
 ROOT := $(shell pwd)
```

**Comment by rkd-msw** [comment]
I think this is a duplicate of https://github.com/Metaswitch/chronos/issues/208#issuecomment-166156137 - @bossmc , in that issue, you suggested that running `make submodules` first was the intended workflow. Should @dtheobald be running `make submodules && make build` or should we be fixing something here?

**Comment by dtheobald** [comment]
I think it should be consistent with the other makefiles.

**Comment by mirw** [comment]
@dtheobald, please can you fix up the details (in particular the summary) of this issue?  The problem is not that it doesn't build, but that its build process differs from other components.

**Comment by bossmc** [comment]
I've updated the development docs in #217 to make the chronos-specific requirement clear.

**Comment by mirw** [comment]
@bossmc, this is still with you.  What are the outstanding actions here?

**Comment by chris-elford-metaswitch** [comment]
This tripped me up today. The submodules are just a dependency and make is supposed to resolve all of the project's dependencies, so it doesn't make sense that I have to run a separate command. If this is required, then it should be flagged explicitly in the README.

**Comment by eleanor-merry** [comment]
It's in the development doc (https://github.com/Metaswitch/chronos/pull/217), and @rkd-msw is fixing the structure up any way in https://github.com/Metaswitch/chronos/pull/222

**Comment by chris-elford-metaswitch** [comment]
But I didn't see any link to the development doc from the README. I had no idea it existed until Matt pointed me at it.

**Comment by chris-elford-metaswitch** [comment]
Thanks @rkd-msw.

**Comment by rkd-msw** [comment]
You're welcome @chris-elford-metaswitch

**Comment by rkd-msw** [comment]
Fixed by #222.
